File: smp-backend/go-cron-service/models/attendance.go
package models

// Attendance represents attendance data
type Attendance struct {
	AttendanceID int    `json:"attendance_id" db:"attendance_id"`
	Date         string `json:"date,omitempty" db:"date"`
	Status       string `json:"status,omitempty" db:"status"`
	ClassID      int    `json:"class_id" db:"class_id"`
	SubjectID    int    `json:"subject_id" db:"subject_id"`
	StudentID    int    `json:"student_id" db:"student_id"`
}

File: smp-backend/rust-api/src/models/attendance.rs
use serde::{Serialize, Deserialize};
use sqlx::FromRow;

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Attendance {
    pub attendance_id: u32, // Primary key
    pub date: String,
    pub status: String,
    pub class_id: u32,
    pub subject_id: u32,
    pub student_id: u32, 
} 